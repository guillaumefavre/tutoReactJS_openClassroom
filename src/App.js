import React, { Component } from 'react'
import './App.css'
import GuessCount from './GuessCount'
import Card from './Card'

class App extends Component {

  handleCardClick(card) {
    console.log(card, 'clicked')
  }

  render() {
    const won = new Date().getSeconds() % 2 === 0;
    return (
      <div className="memory">
        <GuessCount guesses={0}/>
        <Card card="A" feedback="hidden" onClickCard={this.handleCardClick}/>
        <Card card="B" feedback="justMatched" onClickCard={this.handleCardClick}/>
        <Card card="C" feedback="justMismatched" onClickCard={this.handleCardClick}/> 
        <Card card="D" feedback="visible" onClickCard={this.handleCardClick}/>
        <Card card="E" feedback="hidden" onClickCard={this.handleCardClick}/>
        <Card card="F" feedback="justMatched" onClickCard={this.handleCardClick}/>
        <Card card="G" feedback="hidden" onClickCard={this.handleCardClick}/>
        {won && <p>Gagn√© !</p>}
      </div>
    )
  }
}

export default App