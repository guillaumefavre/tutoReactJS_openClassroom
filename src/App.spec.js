import React from 'react'
import { shallow } from 'enzyme'
import { expect } from 'chai'
import sinon from 'sinon'
import App, { SYMBOLS } from './App'
import GuessCount from './GuessCount'

describe('<App />', () => {
    it('renders without crashing', () => {
        // Wrapper autour du composant
        const wrapper = shallow(<App/>)
    
        // On vérifie qu'on a un composant à l'intérieur du wrapper
        expect(wrapper).to.contain(<GuessCount guesses={0} />)
    })

    it('has 36 cards', () => {
        const wrapper = shallow(<App/>)

        expect(wrapper.find('Card')).to.have.length(36)
    })

    
    it('should match its reference snapshot', () => {
        const mock = sinon.stub(App.prototype, 'generateCards').returns([...SYMBOLS.repeat(2)])

        try {
            const wrapper = shallow(<App/>)
            expect(wrapper).to.matchSnapshot()
        } finally {
            // Permet de restorer le comportement de generateCards en dehors de ce test
            mock.restore()
        }
    })
    
})