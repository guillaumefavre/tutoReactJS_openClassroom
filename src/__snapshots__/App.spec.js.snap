// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match its reference snapshot 1`] = `
<div
  className="memory"
>
  <GuessCount
    guesses={0}
  />
  <Card
    card="😀"
    feedback="hidden"
    index={0}
    key="0"
    onClickCard={[Function]}
  />
  <Card
    card="🎉"
    feedback="hidden"
    index={1}
    key="1"
    onClickCard={[Function]}
  />
  <Card
    card="💖"
    feedback="hidden"
    index={2}
    key="2"
    onClickCard={[Function]}
  />
  <Card
    card="🎩"
    feedback="hidden"
    index={3}
    key="3"
    onClickCard={[Function]}
  />
  <Card
    card="🐶"
    feedback="hidden"
    index={4}
    key="4"
    onClickCard={[Function]}
  />
  <Card
    card="🐱"
    feedback="hidden"
    index={5}
    key="5"
    onClickCard={[Function]}
  />
  <Card
    card="🦄"
    feedback="hidden"
    index={6}
    key="6"
    onClickCard={[Function]}
  />
  <Card
    card="🐬"
    feedback="hidden"
    index={7}
    key="7"
    onClickCard={[Function]}
  />
  <Card
    card="🌍"
    feedback="hidden"
    index={8}
    key="8"
    onClickCard={[Function]}
  />
  <Card
    card="🌛"
    feedback="hidden"
    index={9}
    key="9"
    onClickCard={[Function]}
  />
  <Card
    card="🌞"
    feedback="hidden"
    index={10}
    key="10"
    onClickCard={[Function]}
  />
  <Card
    card="💫"
    feedback="hidden"
    index={11}
    key="11"
    onClickCard={[Function]}
  />
  <Card
    card="🍎"
    feedback="hidden"
    index={12}
    key="12"
    onClickCard={[Function]}
  />
  <Card
    card="🍌"
    feedback="hidden"
    index={13}
    key="13"
    onClickCard={[Function]}
  />
  <Card
    card="🍓"
    feedback="hidden"
    index={14}
    key="14"
    onClickCard={[Function]}
  />
  <Card
    card="🍐"
    feedback="hidden"
    index={15}
    key="15"
    onClickCard={[Function]}
  />
  <Card
    card="🍟"
    feedback="hidden"
    index={16}
    key="16"
    onClickCard={[Function]}
  />
  <Card
    card="🍿"
    feedback="hidden"
    index={17}
    key="17"
    onClickCard={[Function]}
  />
  <Card
    card="😀"
    feedback="hidden"
    index={18}
    key="18"
    onClickCard={[Function]}
  />
  <Card
    card="🎉"
    feedback="hidden"
    index={19}
    key="19"
    onClickCard={[Function]}
  />
  <Card
    card="💖"
    feedback="hidden"
    index={20}
    key="20"
    onClickCard={[Function]}
  />
  <Card
    card="🎩"
    feedback="hidden"
    index={21}
    key="21"
    onClickCard={[Function]}
  />
  <Card
    card="🐶"
    feedback="hidden"
    index={22}
    key="22"
    onClickCard={[Function]}
  />
  <Card
    card="🐱"
    feedback="hidden"
    index={23}
    key="23"
    onClickCard={[Function]}
  />
  <Card
    card="🦄"
    feedback="hidden"
    index={24}
    key="24"
    onClickCard={[Function]}
  />
  <Card
    card="🐬"
    feedback="hidden"
    index={25}
    key="25"
    onClickCard={[Function]}
  />
  <Card
    card="🌍"
    feedback="hidden"
    index={26}
    key="26"
    onClickCard={[Function]}
  />
  <Card
    card="🌛"
    feedback="hidden"
    index={27}
    key="27"
    onClickCard={[Function]}
  />
  <Card
    card="🌞"
    feedback="hidden"
    index={28}
    key="28"
    onClickCard={[Function]}
  />
  <Card
    card="💫"
    feedback="hidden"
    index={29}
    key="29"
    onClickCard={[Function]}
  />
  <Card
    card="🍎"
    feedback="hidden"
    index={30}
    key="30"
    onClickCard={[Function]}
  />
  <Card
    card="🍌"
    feedback="hidden"
    index={31}
    key="31"
    onClickCard={[Function]}
  />
  <Card
    card="🍓"
    feedback="hidden"
    index={32}
    key="32"
    onClickCard={[Function]}
  />
  <Card
    card="🍐"
    feedback="hidden"
    index={33}
    key="33"
    onClickCard={[Function]}
  />
  <Card
    card="🍟"
    feedback="hidden"
    index={34}
    key="34"
    onClickCard={[Function]}
  />
  <Card
    card="🍿"
    feedback="hidden"
    index={35}
    key="35"
    onClickCard={[Function]}
  />
</div>
`;
